Examples/Swift/UserTrackingModesExample.swift

```diff
@@ -115,26 +120,35 @@ class UserLocationButton : UIButton {
         return bezierPath.cgPath
     }
     
-    func updateArrow(for mode: MGLUserTrackingMode) {
-        var stroke: CGColor
-        switch (mode) {
+    // Update the arrow's color and rotation when
+    // tracking mode is changed
+    func updateArrowForTrackingMode(mode: MGLUserTrackingMode) {
+        let activePrimaryColor = UIColor.red
+        let disabledPrimaryColor = UIColor.clear
+        let disabledSecondaryColor = UIColor.white
+        let rotatedArrow = CGFloat(0.66)
+        
+        switch mode {
         case .none:
-            stroke = UIColor.white.cgColor
+            updateArrow(with: disabledPrimaryColor, strokeColor: disabledSecondaryColor, rotation: 0)
+            break
         case .follow:
-            stroke = tintColor.cgColor
-        case .followWithHeading, .followWithCourse:
-            stroke = UIColor.clear.cgColor
+            updateArrow(with: disabledPrimaryColor, strokeColor: activePrimaryColor, rotation: 0)
+            break
+        case .followWithHeading:
+            updateArrow(with: activePrimaryColor, strokeColor: activePrimaryColor, rotation: rotatedArrow)
+            break
+        case .followWithCourse:
+            updateArrow(with: activePrimaryColor, strokeColor: activePrimaryColor, rotation: 0)
+            break
         }
-        arrow!.strokeColor = stroke
-        
-        // Re-center the arrow, based on its current orientation.
-        arrow!.position = (mode == .none || mode == .followWithCourse) ? CGPoint(x: size / 2, y: size / 2) : CGPoint(x: size / 2 + 2, y: size / 2 - 2)
-        
-        arrow!.fillColor = (mode == .none || mode == .follow) ? UIColor.clear.cgColor : tintColor.cgColor
-        
-        let rotation: CGFloat = (mode == .follow || mode == .followWithHeading) ? 0.66 : 0
-        arrow!.setAffineTransform(CGAffineTransform.identity.rotated(by: rotation))
-        
+    }
+    
+    func updateArrow(with fillColor: UIColor, strokeColor: UIColor, rotation: CGFloat) {
```

Comment:
`with` in this method ends up obscuring what the first parameter does, so I think we should remove it.